express/app.js

```diff
@@ -0,0 +1,142 @@
+const express = require('express');
+const nunjucks = require('nunjucks');
+const fs = require("fs");
+const path = require("path");
+const AWS = require('aws-sdk');
+AWS.config.update({region:'eu-west-1'});
+
+const encrypter = require('./modules/kms-encrypter');
+const heartbeat = require('./modules/aws-heartbeat');
+const app = express();
+const port = 3000;
+
+const env = nunjucks.configure('templates', {
+    express: app,
+    autoescape: true
+});
+
+env.addFilter('datetime', function(str) {
+    let match = /^(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})/.exec(str);
+    let dateObject = new Date(match[1], match[2]-1, match[3], match[4], match[5], match[6]);
+    let options = {
+        hour12: false,
+        year: 'numeric',
+        month: 'short',
+        day: 'numeric',
+        hour: '2-digit',
+        minute: '2-digit',
+        second: '2-digit'
+    };
+    return dateObject.toLocaleString("en-GB", options);
+});
+app.set('view engine', 'html');
+
+app.use('/assets', express.static('assets'));
+
+/*
+In the interests of encryption you run the express server
+through aws-vault or similar so you can only render audit
+results for the current account assumed through aws-vault
+ */
+//app.get('/', async (req, res) => {
+//    const directoryPath = path.join(__dirname, '../cli/results');
+//    //passsing directoryPath and callback function
+//    let files = await fs.readdir(directoryPath, function (err, files) {
+//        let accounts = [], i = 0;
+//        for (i=0; i<files.length;i++) {
+//            if (files[i].match(/^\d{12}$/)) {
+//                accounts.push(files[i]);
+//            }
+//        }
+//        accounts.sort();
+//        res.render('cli_account_list.html', {
+//            title: 'Acoounts: Cloud Security Watch',
+//            asset_path: '/assets',
+//            accounts: accounts
+//        });
+//    });
+//
+//});
+
+app.get('/', async (req, res) => {
+    const sts = new AWS.STS();
+    await sts.getCallerIdentity({}, async function(err, data) {
+        if (err) {
+            console.log(err, err.stack); // an error occurred
+        } else {
+            let account = data["Account"];
+            const directoryPath = path.join(__dirname, '../cli/results/'+account);
```

Comment:
Is the "../" in the path required if you're joining it to the directory name?

Reply:
The __dirname the cwd which is `csw-backend/express` we need to get to `csw-backend/cli`